test/integration/etcd/etcd_storage_path_test.go

```diff
@@ -695,26 +701,35 @@ func startRealMasterOrDie(t *testing.T, certDir string) (*allClient, clientv3.KV
 				t.Log(err)
 			}
 
-			time.Sleep(100 * time.Millisecond)
+			time.Sleep(time.Second)
 		}
 	}()
 
-	if err := wait.PollImmediate(100*time.Millisecond, 10*time.Second, func() (done bool, err error) {
+	if err := wait.PollImmediate(time.Second, time.Minute, func() (done bool, err error) {
 		obj := kubeClientConfigValue.Load()
 		if obj == nil {
 			return false, nil
 		}
 		kubeClientConfig := kubeClientConfigValue.Load().(*restclient.Config)
-		kubeClient, err := kclient.NewForConfig(kubeClientConfig)
+		// make a copy so we can mutate it to set GroupVersion and NegotiatedSerializer
+		cfg := *kubeClientConfig
+		cfg.ContentConfig.GroupVersion = &schema.GroupVersion{}
+		cfg.ContentConfig.NegotiatedSerializer = kapi.Codecs
+		privilegedClient, err := restclient.RESTClientFor(&cfg)
```

Comment:
Split the health check call logic into a separate function? It doesn't need to be inline here.

Reply:
This cannot be split up because it uses `atomic.Value` which is not safe to copy.